inference-engine/src/mkldnn_plugin/nodes/mkldnn_adaptive_pooling.cpp

```diff
@@ -0,0 +1,363 @@
+                case 3:
+                default:
+                    indexFmt = memory::format_tag::ncdhw;
+            }
+            config.outConfs[1].desc = MKLDNNMemoryDesc(getChildEdgeAt(1)->getDims(), memory::data_type::s32, indexFmt);
+        }
+        supportedPrimitiveDescriptors.push_back({config, impl_desc_type::unknown, fmt});
+    }
+}
+
+std::vector<memory::format_tag> MKLDNNAdaPoolingNode::getAvailableFormatsForDims(const MKLDNNDims &dims) const {
+    auto plainFmt = [&]() {
+        if (dims.ndims() == 3)
+            return memory::format_tag::ncw;
+        else if (dims.ndims() == 4)
+            return memory::format_tag::nchw;
+        else if (dims.ndims() == 5)
+            return memory::format_tag::ncdhw;
+        return memory::format_tag::any;
+    };
+    auto blockedFmt = [&]() {
+        if (dims.ndims() == 3)
+            return mayiuse(avx512_common) ? memory::format_tag::nCw16c : memory::format_tag::nCw8c;
+        else if (dims.ndims() == 4)
+            return mayiuse(avx512_common) ? memory::format_tag::nChw16c : memory::format_tag::nChw8c;
+        else if (dims.ndims() == 5)
+            return mayiuse(avx512_common) ? memory::format_tag::nCdhw16c : memory::format_tag::nCdhw8c;
+        return memory::format_tag::any;
+    };
+    auto tailFmt = [&]() {
+        if (dims.ndims() == 3)
+            return memory::format_tag::nwc;
+        else if (dims.ndims() == 4)
+            return memory::format_tag::nhwc;
+        else if (dims.ndims() == 5)
+            return memory::format_tag::ndhwc;
+        return memory::format_tag::any;
+    };
+
+    if (getParentEdgeAt(0)->getDims()[1] == 1) {
+        return { plainFmt() };
+    } else {
+        return { plainFmt(), tailFmt(), blockedFmt() };
+    }
+}
+
+namespace {
+struct AdaPoolingContext {
+    MKLDNNAdaPoolingNode &node;
+};
+}
+
+template<typename T>
+struct MKLDNNAdaPoolingNode::AdaPoolingExecute {
+    using srcT = typename std::tuple_element<0, T>::type;
+    using dstT = typename std::tuple_element<1, T>::type;
+
+    void operator()(AdaPoolingContext & ctx) {
+        ctx.node.executeSpecified<srcT, dstT>();
+    }
+};
+void MKLDNNAdaPoolingNode::execute(mkldnn::stream strm) {
+    auto inputPrec = getParentEdgeAt(0)->getMemory().GetDescriptor().data.data_type;
+    auto outputPrec = getChildEdgeAt(0)->getMemory().GetDescriptor().data.data_type;
+    if (!((inputPrec == mkldnn_bf16 && outputPrec == mkldnn_bf16) ||
+          (inputPrec == mkldnn_f32 && outputPrec == mkldnn_f32)))
+        IE_THROW() <<"AdaPooling doesn't support demanded precisions";
+
+    AdaPoolingContext ctx = {
+            *this
+    };
+
+    OV_SWITCH(MKLDNNPlugin, AdaPoolingExecute, ctx, std::tie(inputPrec, outputPrec),
+              OV_CASE2(mkldnn_f32, mkldnn_f32, float, float),
+              OV_CASE2(mkldnn_bf16, mkldnn_bf16, bfloat16_t, bfloat16_t))
+}
+
+template <typename inputType, typename outputType>
+void MKLDNNAdaPoolingNode::executeSpecified() {
```

Comment:
This implementation does not support BF16. Remove BF16 conditions above.
Also this function should not be templated due to inputType == outputType == FP32.

Reply:
Done

Reply:
Not  fixed. Just one type is expected here. No need to use template here.